### DISCUSSION TITLE
How to provide synchronous responses in controllers?

### DISCUSSION THREAD

**Discussion by mikebronner** [spatie/laravel-event-sourcing#342] [Q&A]
I'm just starting my event-sourcing adventure and am trying to wrap my head how to implement them in API controllers.

1. An API request comes in.
2. We run an action and fire off an event.
3. That event triggers a reactor, which executes an action, and fires off an event.
4. The result of that second action has the information I need to return from the controller.

How do you handle scenarios like this, where there are multiple events in the chain between receiving the request, and responding to it, since events carry information one way.

**Comment by bmbBAMBUS** [comment]
I presume you use internal async queue for events, so there are two options to manage it:
 - return to frontend unique request_uuid and create additional controller which will return desired data whenever it's ready. Use long pooling from frontend.
 - You need to use notifications and push notification with data to frontend. You can use pusher or something similar.
